nucliadb_vectors/src/data_point_provider/merger.rs

```diff
@@ -17,26 +17,52 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-use std::sync::mpsc::{channel, Receiver, Sender};
-use std::sync::Mutex;
+use std::sync::mpsc::{self, Receiver, Sender};
+use std::sync::Once;
 
-use std::thread;
+use crate::{VectorErr, VectorR};
 
-use crate::VectorR;
 pub type MergeRequest = Box<dyn MergeQuery>;
 pub type MergeTxn = Sender<MergeRequest>;
 
 pub trait MergeQuery: Send {
     fn do_work(&self) -> VectorR<()>;
 }
 
-struct Updater {
+#[derive(Clone)]
+struct MergerHandle(MergeTxn);
+impl MergerHandle {
+    pub fn send(&self, request: MergeRequest) {
+        let Err(e) = self.0.send(request) else { return };
+        tracing::info!("Error sending merge request, {e}");
+    }
+}
+
+static mut MERGER_NOTIFIER: Option<MergerHandle> = None;
+static MERGER_NOTIFIER_SET: Once = Once::new();
+
+pub fn send_merge_request(request: MergeRequest) {
+    match unsafe { &MERGER_NOTIFIER } {
```

Comment:
Add a SAFETY comment

Reply:
The function itself is safe so I dont think a safety comment should be added in the sense the [guidelines](https://std-dev-guide.rust-lang.org/documentation/safety-comments.html) specify. But  at [d846d76](https://github.com/nuclia/nucliadb/pull/600/commits/d846d76971f64cd8d803c4562c05718265a55ea3) I have included a comment explaining why the unsafe block has no risks.